src/openrct2/interface/StdInOutConsole.cpp

```diff
@@ -134,7 +134,23 @@ void StdInOutConsole::WriteLine(const std::string& s, FormatToken colourFormat)
     {
         if (_isPromptShowing)
         {
-            std::printf("\r%s%s\x1b[0m\x1b[0K\r\n", formatBegin.c_str(), s.c_str());
+            auto* mainString = s.c_str();
+
+            // If string contains \n, we need to replace with \r\n
+            std::string newString;
+            if (s.find('\n') != std::string::npos)
+            {
+                for (auto ch : s)
+                {
+                    if (ch == '\n')
+                        newString += "\r\n";
+                    else
+                        newString += ch;
+                }
+                mainString = newString.c_str();
```

Comment:
It seems **very** weird, to replace an `std::string` internal `char*`. Does it need to? It created the `newString` already

Reply:
It would only be a `std::string*` otherwise, and would mean you just need a c_str in the WriteLine instead. I don't see any advantage over that way or this way.